src/rgw/rgw_notify.cc

```diff
@@ -694,27 +697,32 @@ void populate_event_from_request(const reservation_t& res,
   boost::algorithm::hex((const char*)&ts, (const char*)&ts + sizeof(utime_t), 
           std::back_inserter(event.object_sequencer));
   set_event_id(event.id, etag, ts);
-  event.bucket_id = s->bucket->get_bucket_id();
-  // pass metadata
-  if (res.cached_metadata.empty()) {
+  event.bucket_id = res.bucket->get_bucket_id();
+  // pass meta data
+  if (res.x_meta_map.empty()) {
     // no metadata cached:
     // either no metadata exist or no metadata filter was used
-    event.x_meta_map = s->info.x_meta_map;
-    metadata_from_attributes(s, obj, event.x_meta_map);
+    metadata_from_attributes(res, obj);
   } else {
-    event.x_meta_map = std::move(res.cached_metadata);
+      event.x_meta_map = res.x_meta_map;
   }
   // pass tags
-  if (s->tagset.get_tags().empty()) {
+  if (!res.tagset ||
+      (*res.tagset).get_tags().empty()) {
     // try to fetch the tags from the attributes
-    tags_from_attributes(s, obj, event.tags);
+    tags_from_attributes(res, obj, event.tags);
   } else {
-    event.tags = s->tagset.get_tags();
+    if (res.tagset) {
```

Comment:
is this "if" needed?
isn't tagset not-null in the above "else"?

Reply:
you are correct